hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/OMClientRequestUtils.java

```diff
@@ -47,4 +60,40 @@ public static void checkClientRequestPrecondition(
           OMException.ResultCodes.INTERNAL_ERROR);
     }
   }
+
+  public static boolean isSnapshotBucket(OMMetadataManager omMetadataManager,
+      OmKeyInfo keyInfo) throws IOException {
+
+    boolean status;
+    TableIterator<String, ? extends Table.KeyValue<String, SnapshotInfo>>
+        iterator = omMetadataManager.getSnapshotInfoTable().iterator();
+    String dbSnapshotBucketKey = omMetadataManager.getBucketKey(
+        keyInfo.getVolumeName(), keyInfo.getBucketName())
+        + OM_KEY_PREFIX;
+
+    iterator.seek(dbSnapshotBucketKey);
+    status = iterator.hasNext() && iterator.next().getKey()
+        .startsWith(dbSnapshotBucketKey);
+
+    return status || checkSnapshotCache(omMetadataManager, keyInfo);
+  }
+
+  private static boolean checkSnapshotCache(OMMetadataManager omMetadataManager,
```

Comment:
```suggestion
  private static boolean checkSnapshotInCache(OMMetadataManager omMetadataManager,
      OmKeyInfo keyInfo) 
```

Or `checkInSnapshotCache` whatever is more relevant. `checkSnapshotCache` is vague to me.